sdk/core/azure-core/azure/core/pipeline_client_async.py

```diff
@@ -82,16 +76,14 @@ class AsyncPipelineClient(PipelineClientBase):
             :caption: Builds the async pipeline client.
     """
 
-    def __init__(self, base_url, config, **kwargs):
+    def __init__(self, base_url, **kwargs):
         super(AsyncPipelineClient, self).__init__(base_url)
-        if config is None:
-            raise ValueError("Config is a required parameter")
-        self._config = config
+        self._config = kwargs.get("config", None) or Configuration(**kwargs)
         self._base_url = base_url
         if kwargs.get("pipeline"):
             self._pipeline = kwargs["pipeline"]
         else:
-            self._pipeline = self._build_pipeline(config, **kwargs)
+            self._pipeline = self._build_pipeline(self._config, **kwargs)
```

Comment:
same suggestion as above, don't store `self._config` if there is not actually any need